raft/node_test.go

```diff
@@ -190,6 +190,44 @@ func TestNodeReadIndex(t *testing.T) {
 	}
 }
 
+// TestNodeReadIndexToOldLeader ensures that raftpb.MsgReadIndex to old leader
+// gets forwarded to the new leader and 'send' method does not attach its term.
+func TestNodeReadIndexToOldLeader(t *testing.T) {
```

Comment:
probably all we want to test here is:

- elect 1 as leader

- send readindex request to 2

- verify 2 forwards this message to 1 with term not set

- send readindex request to 3

- verify 3 forwards this message to 1 with term not set as well.

- now elect 3 as leader

- let 1 steps the two messages previously we got from 2, 3

- verify 1 forwards these messages again to the new leader 3